R/resourceControl.R

```diff
@@ -2,28 +2,63 @@
 #' @inheritParams abundanceControl
 resourceControl <- function(input, output, session, params, flags, ...) {
     observe({
+        re <- isolate(input$re) # to know which background is selected
+        p <- isolate(params())
         req(input$kappa,
             input$lambda,
             input$log_r_pp,
             input$w_pp_cutoff,
             input$n_resource)
-        p <- setResource(isolate(params()),
+        if(!is.null(getComponent(p, "MR")))
+        {
+            no_re <- which(p@other_params$other$MR$resource_params$resource == re)
```

Comment:
I inherited from early versions of mizer the unfortunate convention that `no_` stands for "number of", so `no_w` is the number of size bins, `no_sp` is the number of species, and so on. So I would prefer `sel_re` for a logical selector like this.